### PR TITLE
TSCH port for CC2538 based platforms.

### PR DESCRIPTION
**Pull request by sumanpanchal**
Have tested rpl-tsch example using two openmote-cc2538. It works.!! :)

### CODE REVIEW

**Comment by simonduq** [comment]
how about the readme file? ;)

**Comment by simonduq** [comment]
(I'm fine if you simplify and use "cc2538-based platforms")

**Comment by simonduq** [comment]
:+1:

File: platform/openmote-cc2538/contiki-conf.h [hunk 1]
```diff
@@ -79,6 +79,14 @@ typedef uint32_t rtimer_clock_t;
 #define RTIMER_CLOCK_DIFF(a, b)     ((int32_t)((a) - (b)))
 /** @} */
 /*---------------------------------------------------------------------------*/
+#define TSCH_CONF_HW_FRAME_FILTERING    0
```
Comment (simonduq) [inline platform/openmote-cc2538/contiki-conf.h]: Do we want this set unconditionally? If needed only with TSCH let's add it to the list of platform-specific things we need to add for TSCH, see project-conf of rpl-tsch example. We need to make this configuration more smooth later but that is a different topic.

**Comment by g-oikonomou** [comment]
@sumanpanchal where are we on this one? This seems pretty straightforward considering that the RF driver supports TSCH in the first place.

@peretuset any thoughts here?

**Comment by peretuset** [comment]
@g-oikonomou, I agree with @simonduq that we should simply state cc2538-based platforms since all of them will support TSCH out of the box once the driver supports it. Do we want to use this PR to do it? Or shall we open a new one?

**Comment by g-oikonomou** [comment]
No strong opinion, may as well use this one if suitable.

**Comment by simonduq** [comment]
Fine either way, but the PR needs a rebase now

**Comment by alignan** [comment]
Rename the PR and make the change for all cc2538 platforms, no need to further delay this

**Comment by peretuset** [comment]
@sumanpanchal, please follow @alignan suggestion in order to complete the PR.

**Comment by sumanpanchal** [comment]
Dear All,
I have done all the necessary changes as per suggestion. I am unable to change PR title and branch name. Please let me know how to do that.

**Comment by simonduq** [comment]
Hello,
There still needs to be some conflicts, have rebased on the latest master?

**Comment by sumanpanchal** [comment]
@simonduq 
I have rebased on the latest master now.

File: examples/ipv6/rpl-tsch/project-conf.h [hunk 2]
```diff
@@ -164,4 +164,9 @@
 
 #endif /* CONTIKI_TARGET_Z1 */
 
+#if CONTIKI_TARGET_CC2538DK || CONTIKI_TARGET_ZOUL || \
+    CONTIKI_TARGET_OPENMOTE_CC2538
+#define TSCH_CONF_HW_FRAME_FILTERING    0
+#endif /* CONTIKI_TARGET_CC2538DK || CONTIKI_TARGET_ZOUL */
```
Comment (tonnenpinguin) [inline examples/ipv6/rpl-tsch/project-conf.h]: Minor issue, but maybe you could update (or remove) the comment to reflect all three defines

**Comment by alignan** [comment]
Thanks @sumanpanchal!

**Comment by peretuset** [comment]
Thanks!

**Comment by sumanpanchal** [comment]
Welcome :smile:

**Pull request merged**